### DISCUSSION TITLE
Change app icon

### DISCUSSION THREAD

**Discussion by mflowtron** [buildnatively/community#64] [Ideas]
### Use case

```Text
Apps like Trainerize allow their users certain levels of branding such as the ability to change the app icon. This is used in white labeling situations.
```

### Idea

```Text
A Workflow Action that allows you to change the image used for the app icon for that specific device.
```

**Comment by romanfurman6** [comment]
Unfortunately, the idea you submitted will not work for iOS. You need to upload static images inside a build and then change them. Please let me know if this is what you're looking for.

**Comment by mflowtron** [comment]
I recorded a quick screen capture of the Trainerize app doing this just to make sure we're on the same page.

https://github.com/buildnatively/community/assets/125224036/1a5f4710-a5de-47e2-beef-0ffcc311cb6b

That's really interesting! I know that in the case of Trainerize they have dozens or more different icons depending on the white label partner. So they would need to release a new build to the App Store every time a new white label partner is added in order to add this feature?

**Reply by romanfurman6** [reply]
Ye it should be static, probably they release app updates
https://www.hackingwithswift.com/example-code/uikit/how-to-change-your-app-icon-dynamically-with-setalternateiconname

![SCR-20240110-lgi](https://github.com/buildnatively/community/assets/22365403/4ef51088-0e0a-4e09-afc6-ae614a4e4b3b)
